### PR TITLE
Add help text to sui start for ProtocolConfig overrides

### PR DESCRIPTION
**Pull request by aschran**

### CODE REVIEW

File: crates/sui/src/sui_commands.rs [hunk 1]
```diff
@@ -147,7 +147,18 @@ pub enum SuiCommand {
...
     ///
     /// Note that if you want to start an indexer, Postgres DB is required.
-    #[clap(name = "start")]
+    ///
+    /// ProtocolConfig parameters can be overridden individually by setting env variables as
+    /// follows:
+    /// - SUI_PROTOCOL_CONFIG_OVERRIDE_ENABLE=1
+    /// - Then, to configure an override, use the prefix `SUI_PROTOCOL_CONFIG_OVERRIDE_`
+    ///   along with the parameter name. For example, to reduce the rate of checkpoint
+    ///   creation to <1/s, you might set:
```
[the remark this answers is not in the document]
Comment (aschran) [reply crates/sui/src/sui_commands.rs:156]: I wouldn't say "this is the only use case", you can use it for whatever you want. This is the only thing I personally have seen requested, that's quite a different statement from "this is the only use case"

[the remark this answers is not in the document]
Comment (aschran) [reply crates/sui/src/sui_commands.rs]: good point, reworded

**Review by mario4tier — COMMENTED** [review]
Rewording review done. Thanks.

**Comment by aschran** [comment]
If one of you wants to approve this I can merge it

**Review by stefan-mysten — APPROVED** [review]

**Comment by stefan-mysten** [comment]
@aschran I approved this as it looked low risk to me, but let me know if there's anything that I missed.

**Pull request merged**